Add Responsive Footer

chidexebere opened the pull request:
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Fixes #725  by @panchovm 

## Description
<!-- Concisely describe what the pull request does. -->
This adds a responsive footer with styling as specified in the Design Library for different screen views.

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

obulat reviewed (changes requested):
Looks great! Just a couple of points to change:
The donate button should have a transparent border, event when hovered / focused.
Locale's background color should be light gray...

chidexebere commented:
> Looks great! Just a couple of points to change:
> The donate button should have a transparent border, event when hovered / focused.
> Locale's background color should be light gray...

- There is no hover/focus state for the footer under the figma DL showing any state of the donate button. I feel changes to the donate button should be done separately on a different issue since the donate button can be used in another composite component.

- There is no locale for the footer under the DL.

zackkrida commented:
@obulat Are your comments here for the header PR, any chance? Your comment about the locale chooser made me suspicious.

obulat commented on packages/vocabulary/src/stories/header.stories.mdx:
```diff
@@ -8,7 +8,7 @@ import { Meta, Preview, Story } from '@storybook/addon-docs/blocks'
       <nav class="navbar ${size}">
-        <div class="navbar-brand">
+        <div class="navbar-brand logo">
           <a class="has-text-black">
```
Could you add an 'href' to links ("/" for this one, and maybe "#0" for others)? This would help us test focus styles and tabbing. Currently, you can't navigate the header using tabs.

obulat commented on packages/vocabulary/src/styles/footer.scss:
```diff
@@ -1,131 +1,289 @@
+		}
+		p {
+			font-size: 0.813rem;
+			font-weight: 600;
```
Font weight should be normal.

chidexebere replied:
Please check DL, font weight is 600 
![donate](https://user-images.githubusercontent.com/25040059/99304479-b8ee8200-2852-11eb-8ddd-0f527f64df52.png)

obulat replied:
Oh, I wasn't looking at the mobile version. It's 600 on mobile, and normal on other versions.

chidexebere replied:
ohhh, true......fixed

obulat commented on packages/vocabulary/src/styles/footer.scss:
```diff
@@ -1,131 +1,289 @@
+			color: $color-white;
+			font-family: $family-heading;
+			text-transform: uppercase;
+			width: 70%;
```
On tablet widths, the width should be 100%

chidexebere replied:
true , fixed

obulat reviewed (changes requested):
The header looks great! I added only two tiny points in the comments.
The footer is great, too. It needs some more work on responsiveness: it currently breaks at widths 1024px and 320px.

obulat commented:
I think I finally understood why I'm confused! @chidexebere, did you base your footer branch off of your header branch? Currently, it looks like you've changed both header and footer in this PR, so I reviewed both ...  So, never mind my remarks about the header!

chidexebere commented:
> The header looks great! I added only two tiny points in the comments.
> The footer is great, too. It needs some more work on responsiveness: it currently breaks at widths 1024px and 320px.

Fixed the issue with the menu-items breaking at 1024px. For below 320px, this is majorly because bulma elements have a `min-width` of 300px.  Most mobile phones do not have screens less than 320px in width and since we are not building for wearable, I felt it was not necessary. But we can get inputs from @zackkrida , @dhruvkb , @Dhruvi16  and @hugosolar  on this.

chidexebere commented:
> I think I finally understood why I'm confused! @chidexebere, did you base your footer branch off of your header branch? Currently, it looks like you've changed both header and footer in this PR, so I reviewed both ... So, never mind my remarks about the header!

No, I did not make this branch off my header branch neither did I make any changes to the header. You can review the `files changed`

obulat commented:
> > I think I finally understood why I'm confused! @chidexebere, did you base your footer branch off of your header branch? Currently, it looks like you've changed both header and footer in this PR, so I reviewed both ... So, never mind my remarks about the header!
> 
> No, I did not make this branch off my header branch neither did I make any changes to the header. You can review the `files changed`

This is what I saw when I clicked on 'Changes since your last review': 
![2020-11-17_08-32-47](https://user-images.githubusercontent.com/15233243/99350510-6bbee000-2853-11eb-88ed-e4233658ee7d.png)
@chidexebere , Now when I look at the files changed I can see only changes to the footer, so whatever the problem was, it's fixed :)

obulat commented:
@chidexebere , we updated the button styles to match DL and merged yesterday. So, when you use `.button` now, the correct font and text-transform are set automatically. For the footer, you can use `.button` for the subscribe button, and `.button.donate.medium` for the donate button, and remove all unnecessary styles. Also, please add  `--button-focus-outline: white;` to the top of the footer styles. This will make sure that all buttons in the footer have a white outline when focused, for accessibility purposes.

obulat commented:
> > The header looks great! I added only two tiny points in the comments.
> > The footer is great, too. It needs some more work on responsiveness: it currently breaks at widths 1024px and 320px.
> 
> Fixed the issue with the menu-items breaking at 1024px. For below 320px, this is majorly because bulma elements have a `min-width` of 300px. Most mobile phones do not have screens less than 320px in width and since we are not building for wearable, I felt it was not necessary. But we can get inputs from @zackkrida , @dhruvkb , @Dhruvi16 and @hugosolar on this.

I think the smallest size I looked at yesterday was 320px, and there the subscribe form was wider than the screen, but you've fixed it now. I don't think we should be adding screens narrower than that.
Responsiveness looks great on all widths now!

chidexebere commented:
> @chidexebere , we updated the button styles to match DL and merged yesterday. So, when you use `.button` now, the correct font and text-transform are set automatically. For the footer, you can use `.button` for the subscribe button, and `.button.donate.medium` for the donate button, and remove all unnecessary styles. Also, please add `--button-focus-outline: white;` to the top of the footer styles. This will make sure that all buttons in the footer have a white outline when focused, for accessibility purposes.

Okay......styles updated
